packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,138 @@ void main() {
     expect(value, equals('three'));
   });
 
+  testWidgets('DropdownButtonFormFieldState value updates when a different option is selected', (WidgetTester tester) async {
+    const String value = 'one';
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: value,
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // [FormField] value is properly initialized.
+    expect(stateKey.currentState.value, equals('one'));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('one'));
+    await tester.pumpAndSettle();
+
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+  });
+
+  testWidgets("DropdownButtonFormField's InputDecorator isEmpty behavior is correct", (WidgetTester tester) async {
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: null,
```

Comment:
nit: explain how to set `isEmpty` to true.
```suggestion
            value: null, // setting value to null sets isEmpty to true
```